apps/test/unit/lib/levelbuilder/lesson-editor/UploadImageDialogTest.jsx

```diff
@@ -38,14 +41,27 @@ describe('UploadImageDialog', () => {
         uploadImage={uploadImage}
       />
     );
-    wrapper.setState({imgUrl: 'http://example.com/img.png'});
+    const returnData = {newAssetUrl: 'http://example.com/img.png'};
+    const fetchStub = sinon
+      .stub(window, 'fetch')
+      .returns(Promise.resolve({ok: true, json: () => returnData}));
 
+    wrapper
+      .find('input')
+      .first()
+      .simulate('change', {target: {files: ['filedata']}});
     expect(handleClose.callCount).to.equal(0);
     expect(uploadImage.callCount).to.equal(0);
-    wrapper.instance().handleCloseAndSave();
 
-    expect(handleClose.callCount).to.equal(1);
-    expect(uploadImage.callCount).to.equal(1);
-    expect(uploadImage.calledWith('http://example.com/img.png')).to.equal(true);
+    return new Promise(resolve => setImmediate(resolve)).then(() => {
```

Comment:
Found this pattern at https://www.pluralsight.com/guides/testing-asynchronous-functionality-react-component, as well as a couple stackoverflow answers.